core/src/main/scala/io/github/ghostbuster91/sttp/client3/json/circe/CirceCoproductCodecGenerator.scala

```diff
@@ -17,70 +18,72 @@ private[circe] class CirceCoproductCodecGenerator() {
   private def encoder(coproduct: Coproduct): IM[Defn.Val] = {
     val coproductType = coproduct.typeName
     val encoderName = coproduct.asPrefix("Encoder")
-    coproduct.discriminator match {
-      case Some(discriminator) if discriminator.mapping.nonEmpty =>
-        for {
-          jsonTpe <- CirceTypeProvider.AnyType
-          encoderTpe <- CirceTypeProvider.EncoderTpe
-        } yield {
-          val cases = encoderCasesWithMapping(discriminator)
-          val encoderInit = init"${t"$encoderTpe[$coproductType]"}()"
-          q"""implicit lazy val $encoderName: $encoderTpe[$coproductType] = new $encoderInit {
-            override def apply(${coproduct.toVar}: $coproductType): $jsonTpe = 
-              ${coproduct.toVar} match {
-                ..case $cases
-                }
-            }
-            """
-        }
-      case _ =>
-        for {
-          encoderTpe <- CirceTypeProvider.EncoderTpe
-        } yield {
-          val cases = encoderCases(coproduct)
-          val encoderInit = init"${t"$encoderTpe[$coproductType]"}()"
-          q"""implicit lazy val $encoderName: $encoderTpe[$coproductType] = Encoder.instance {
-                ..case $cases
-            }
-            """
+    for {
+      encoderTpe <- CirceTypeProvider.EncoderTpe
+    } yield {
+      val cases = encoderCases(coproduct)
+      val encoderInit = init"${t"$encoderTpe[$coproductType]"}()"
+      q"""implicit lazy val $encoderName: $encoderTpe[$coproductType] = Encoder.instance {
+            ..case $cases
         }
+        """
```

Comment:
I think that it is fine cause when you use enum for a discriminator you probably want it to enumerate all the cases, otherwise what would be the point of using enum, right?